xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/listener/BackLinkUpdaterListener.java

```diff
@@ -80,6 +84,9 @@ public class BackLinkUpdaterListener extends AbstractLocalEventListener
     @Inject
     private JobContext jobContext;
 
+    @Inject
+    private LinkStore linkStore;
```

Comment:
Might cause an initialization of the link store too early, depending on the implementation (for example it was a no go back when it was based on the database, it should theoretically work OK with the Solr one I think, but would probably still be better to not initialize a whole store just because of the listener which might not be used before long) so it would probably be safer to use a Provider<LinkStore> here.